### ISSUE TITLE
`extrude` method to extrude geometries in the Z direction

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/GeometryOps.jl#136]
Ideally this would work for points, linestrings, and polygons, but right now we can restrict it to only working on polygons.  

The idea would be to take a polygon, mesh it, and "extrude" it in the z-axis to create a 3d mesh.  We would also need a triangulation interface first, but that seems fairly easy to do, and there are plenty of packages which would do it.

This would make creating plots like the one below easier (yes it is bad practice, but it's _cool_)
<img width="442" alt="Screenshot 2024-05-06 at 2 01 01 PM" src="https://github.com/JuliaGeo/GeometryOps.jl/assets/32143268/85a3d9ec-0a76-4614-b466-b7d1f16e35b4">
Is there any reason we should not do this?

**Comment by rafaqz** [comment]
Does it need a mesh? Not just 3d polygons?

**Comment by asinghvi17** [comment]
Yeah, the sides have to be meshes and triangulation happens in ℝ² unless you do it yourself
